### DISCUSSION TITLE
with the proto3, the oneof cannot be set.

### DISCUSSION THREAD

**Discussion by steed717** [protobuf-c/protobuf-c#582] [Q&A] [answered]
Hi, with the protobuf 3, the reciever side can not get the right one of value. The reciever side can only get the default UNKONWN value.

**Comment by lipnitsk** [comment, marked as answer]
Make sure you set the auto-generated `case` enum to the correct case, otherwise protobuf-c does not know which field changed. Please use the mailing list for questions and not open issues for questions that are not bugs.
